### PR TITLE
FIX-#4756: Correctly propagate `storage_options` in `read_parquet`

### LINKED ISSUE #4756: Modin issue to cause the read_parquet file from Minio with error
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **Modin version** (`modin.__version__`):
- **Python version**:
- **Code we can use to reproduce**:

### Describe the problem
Error:  i can use the latest modin to read_csv from Modin/Minio and can not read_parquet from Modin/Minio with below error message. But if i used with standard pandas i still can read_parquet from Minio without modin. this is the Modin problem with minio and read_parquet function.

Error message as  below:

<details>

<summary>Stack trace</summary>

```
Caught schedule exception
Caught schedule exception
Caught schedule exception
Caught schedule exception
Caught schedule exception
Caught schedule exception
Caught schedule exception
Caught schedule exception
Caught schedule exception
Caught schedule exception
Caught schedule exception
Caught schedule exception
Caught schedule exception
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
/tmp/ipykernel_309/3390206043.py in <module>
     42 else:
     43     minio_file_name = 's3://test/part-00017.parquet' #files[i] #
---> 44     df = pd.read_parquet(
     45             #'s3://test/part-00015.parquet',
     46             minio_file_name,

/opt/conda/lib/python3.9/site-packages/modin/logging/logger_decorator.py in run_and_log(*args, **kwargs)
    126             """
    127             if LogMode.get() == "disable":
--> 128                 return obj(*args, **kwargs)
    129 
    130             logger = get_logger()

/opt/conda/lib/python3.9/site-packages/modin/_compat/pandas_api/latest/io.py in read_parquet(path, engine, columns, storage_options, use_nullable_dtypes, **kwargs)
    207 
    208     return DataFrame(
--> 209         query_compiler=FactoryDispatcher.read_parquet(
    210             path=path,
    211             engine=engine,

/opt/conda/lib/python3.9/site-packages/modin/core/execution/dispatching/factories/dispatcher.py in read_parquet(cls, **kwargs)
    178     @_inherit_docstrings(factories.BaseFactory._read_parquet)
    179     def read_parquet(cls, **kwargs):
--> 180         return cls.__factory._read_parquet(**kwargs)
    181 
    182     @classmethod

/opt/conda/lib/python3.9/site-packages/modin/core/execution/dispatching/factories/factories.py in _read_parquet(cls, **kwargs)
    205     )
    206     def _read_parquet(cls, **kwargs):
--> 207         return cls.io_cls.read_parquet(**kwargs)
    208 
    209     @classmethod

/opt/conda/lib/python3.9/site-packages/modin/logging/logger_decorator.py in run_and_log(*args, **kwargs)
    126             """
    127             if LogMode.get() == "disable":
--> 128                 return obj(*args, **kwargs)
    129 
    130             logger = get_logger()

/opt/conda/lib/python3.9/site-packages/modin/core/io/file_dispatcher.py in read(cls, *args, **kwargs)
    154         postprocessing work on the resulting query_compiler object.
    155         """
--> 156         query_compiler = cls._read(*args, **kwargs)
    157         # TODO (devin-petersohn): Make this section more general for non-pandas kernel
    158         # implementations.

/opt/conda/lib/python3.9/site-packages/modin/logging/logger_decorator.py in run_and_log(*args, **kwargs)
    126             """
    127             if LogMode.get() == "disable":
--> 128                 return obj(*args, **kwargs)
    129 
    130             logger = get_logger()

/opt/conda/lib/python3.9/site-packages/modin/core/io/column_stores/parquet_dispatcher.py in _read(cls, path, engine, columns, **kwargs)
    403         ]
    404 
--> 405         return cls.build_query_compiler(path, columns, index_columns, **kwargs)

/opt/conda/lib/python3.9/site-packages/modin/logging/logger_decorator.py in run_and_log(*args, **kwargs)
    126             """
    127             if LogMode.get() == "disable":
--> 128                 return obj(*args, **kwargs)
    129 
    130             logger = get_logger()

/opt/conda/lib/python3.9/site-packages/modin/core/io/column_stores/parquet_dispatcher.py in build_query_compiler(cls, path, columns, index_columns, **kwargs)
    309         col_partitions, column_widths = cls.build_columns(columns)
    310         partition_ids = cls.call_deploy(path, col_partitions, **kwargs)
--> 311         index, sync_index = cls.build_index(path, partition_ids, index_columns)
    312         remote_parts = cls.build_partition(partition_ids, column_widths)
    313         if len(partition_ids) > 0:

/opt/conda/lib/python3.9/site-packages/modin/logging/logger_decorator.py in run_and_log(*args, **kwargs)
    126             """
    127             if LogMode.get() == "disable":
--> 128                 return obj(*args, **kwargs)
    129 
    130             logger = get_logger()

/opt/conda/lib/python3.9/site-packages/modin/core/io/column_stores/parquet_dispatcher.py in build_index(cls, path, partition_ids, index_columns)
    275         if range_index or (len(partition_ids) == 0 and len(column_names_to_read) != 0):
    276             complete_index = (
--> 277                 read_table(path, columns=column_names_to_read).to_pandas().index
    278             )
    279         # Empty DataFrame case

/opt/conda/lib/python3.9/site-packages/pyarrow/parquet/__init__.py in read_table(source, columns, use_threads, metadata, schema, use_pandas_metadata, memory_map, read_dictionary, filesystem, filters, buffer_size, partitioning, use_legacy_dataset, ignore_prefixes, pre_buffer, coerce_int96_timestamp_unit, decryption_properties)
   2735             )
   2736         try:
-> 2737             dataset = _ParquetDatasetV2(
   2738                 source,
   2739                 schema=schema,

/opt/conda/lib/python3.9/site-packages/pyarrow/parquet/__init__.py in __init__(self, path_or_paths, filesystem, filters, partitioning, read_dictionary, buffer_size, memory_map, ignore_prefixes, pre_buffer, coerce_int96_timestamp_unit, schema, decryption_properties, **kwargs)
   2327                     except ValueError:
   2328                         filesystem = LocalFileSystem(use_mmap=memory_map)
-> 2329                 if filesystem.get_file_info(path_or_paths).is_file:
   2330                     single_file = path_or_paths
   2331             else:

/opt/conda/lib/python3.9/site-packages/pyarrow/_fs.pyx in pyarrow._fs.FileSystem.get_file_info()

/opt/conda/lib/python3.9/site-packages/pyarrow/error.pxi in pyarrow.lib.pyarrow_internal_check_status()

/opt/conda/lib/python3.9/site-packages/pyarrow/error.pxi in pyarrow.lib.check_status()

OSError: When getting information for key 'part-00017.parquet' in bucket 'test': AWS Error [code 15]: No response body.


===============================================
```

</details>

### Source code / logs

<details>

<summary>Show source</summary>

```python

 import os
       
        
        MODIN_ENGINE = 'ray'
        RAY_URL = 'ray://ray-ray-head.ray:10001'
        #RAY_URL = 'ray://ray-ray-head.ray.svc.cluster.local:10001'
        AWS_ACCESS_KEY_ID = 'x1miniouser'
        AWS_SECRET_ACCESS_KEY = 'x1miniopass'
        
        import time

        import modin.pandas as pd
        import modin.config as cfg
        
        #import pandas as pd2

        import ray
        
        ray.init(RAY_URL)
        cfg.NPartitions.put(16)        

        select_csv = False
        if select_csv:
            minio_file_name = 's3://test/' + 'df3.csv'
            df = pd.read_csv(
                    #'s3://test/part-00015.parquet',
                    minio_file_name,
                    storage_options={
                        'key': AWS_ACCESS_KEY_ID,
                        'secret': AWS_SECRET_ACCESS_KEY,
                        'client_kwargs': {
                            'endpoint_url': 'https://minio.minio',
                            'verify': False,
                        },
                        'config_kwargs': {
                            'region_name': 'us-west-1',
                            'signature_version': 's3v4',
                        }
                    }
            )
        else:
            minio_file_name = 's3://test/part-00017.parquet' #files[i] #
            df = pd.read_parquet(
                    #'s3://test/part-00015.parquet',
                    minio_file_name,
                    storage_options={
                        'key': AWS_ACCESS_KEY_ID,
                        'secret': AWS_SECRET_ACCESS_KEY,
                        'client_kwargs': {
                            'endpoint_url': 'https://minio.minio',
                            'verify': False,
                        },
                        'config_kwargs': {
                            'region_name': 'us-west-1',
                            'signature_version': 's3v4',
                        }
                    }
            ). 

```

</details>

### PR DESCRIPTION
**Pull request by pyrito**
Signed-off-by: Karthik Velayutham <vkarthik@ponder.io>

## What do these changes do?

This PR addresses a bug where we do not propagate `storage_options` to the `read_table` call in `build_index`. This wasn't caught by any of our CI tests since we didn't stringently check for the options pass in for `read_parquet`. I'm working on getting access to a bucket so we can make sure that this works accordingly. 

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [ ] Resolves #4756 
- [ ] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Comment by YarShev** [comment]
@pyrito, do the changes address the issue?

**Comment by pyrito** [comment]
@YarShev yes they do address the issue. I just need to add a dataset to one of the S3 buckets hosted by the Intel folks with some storage options set so that I know that things are being passed through properly. The existing tests didn't catch this issue.

File: modin/core/io/column_stores/parquet_dispatcher.py [hunk 1]
```diff
@@ -306,9 +334,12 @@ def build_query_compiler(cls, path, columns, index_columns, **kwargs):
         new_query_compiler : BaseQueryCompiler
             Query compiler with imported data for further processing.
         """
+        storage_options = kwargs.pop("storage_options", {}) or {}
```
Comment (YarShev) [inline modin/core/io/column_stores/parquet_dispatcher.py:337]: Why do we pop storage_options of the kwargs?

**Comment by YarShev** [comment]
@pyrito, cool. I left a comment, otherwise LGTM. Please let me know when the changes are ready to be merged.

File: modin/core/io/column_stores/parquet_dispatcher.py [hunk 2]
```diff
@@ -83,7 +83,30 @@ def _unstrip_protocol(protocol, path):
...
+        path : str, path object or file-like object
+            Path to dataset.
+        storage_options : dict
+            Parameters for specific storage engine.
+
+        Returns
+        -------
+        filesystem: Any
+            Protocol implementation of registry.
+        fs_path: list
```
Comment (prutskov) [inline modin/core/io/column_stores/parquet_dispatcher.py:101]:
```suggestion
        filesystem : Any
            Protocol implementation of registry.
        fs_path : list
```

**Review by prutskov — COMMENTED** [review]
LGTM, just a minor suggestion

File: modin/core/io/column_stores/parquet_dispatcher.py (hunk 1 above)
Comment (prutskov) [reply modin/core/io/column_stores/parquet_dispatcher.py:337]: Looks like this line was moved on layer above from `call_deploy` layer to `build_query_compiler`, and `kwargs` which are used in `call_deploy` should not have `storage_options`

**Comment by prutskov** [comment]
@pyrito, please rebase on actual master to make Dask tests passed

File: modin/core/io/column_stores/parquet_dispatcher.py (hunk 1 above)
Comment (pyrito) [reply modin/core/io/column_stores/parquet_dispatcher.py:337]: Yes @prutskov you are correct.

**Review by YarShev — APPROVED** [review]
@pyrito, thanks!

**Pull request merged**